Orange/data/table/impl.py

```diff
@@ -0,0 +1,427 @@
+import warnings
+
+from Orange.data import ContinuousVariable, Domain, StringVariable, DiscreteVariable
+
+from .base import *
+import pandas as pd
+from pandas.sparse.array import BlockIndex
+
+
+class Table(TableBase, pd.DataFrame):
+    """A dense implementation of an Orange table."""
+
+    _KNOWN_PANDAS_KWARGS = {"data", "index", "columns", "dtype", "copy"}
+
+    @property
+    def _constructor(self):
+        """Proper pandas extension as per http://pandas.pydata.org/pandas-docs/stable/internals.html"""
+        return Table
+
+    @property
+    def _constructor_sliced(self):
+        """
+        An ugly workaround for the fact that pandas doesn't transfer _metadata to Series objects.
```

Comment:
[Apparently](https://github.com/pydata/pandas/pull/13978#issuecomment-239946306), this was never a promise. But we need to figure out something else. This `transferer` is so hackish I can't even.